src/services/storage/handlers/stake.ts

```diff
@@ -5,18 +5,261 @@ import { loggingFactory } from '../../../logger'
+      {
+        internalType: 'address',
+        name: 'account',
+        type: 'address'
+      }
+    ],
+    name: 'balanceOf',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  },
+  {
+    constant: false,
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'recipient',
+        type: 'address'
+      },
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    name: 'transfer',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool'
+      }
+    ],
+    payable: false,
+    stateMutability: 'nonpayable',
+    type: 'function'
+  },
+  {
+    constant: true,
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'owner',
+        type: 'address'
+      },
+      {
+        internalType: 'address',
+        name: 'spender',
+        type: 'address'
+      }
+    ],
+    name: 'allowance',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  },
+  {
+    constant: false,
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'spender',
+        type: 'address'
+      },
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    name: 'approve',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool'
+      }
+    ],
+    payable: false,
+    stateMutability: 'nonpayable',
+    type: 'function'
+  },
+  {
+    constant: false,
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'sender',
+        type: 'address'
+      },
+      {
+        internalType: 'address',
+        name: 'recipient',
+        type: 'address'
+      },
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    name: 'transferFrom',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool'
+      }
+    ],
+    payable: false,
+    stateMutability: 'nonpayable',
+    type: 'function'
+  }
+] as AbiItem[]
+
+/**
+ * Native token address
+ */
+export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+/**
+ * Make a call to ERC20 token SC and return token symbol
+ * Return `rbtc` for ZERO_ADDRESS
+ * @param token
+ * @param eth
+ * @param tokenAbi
+ * @returns {SupportedTokens} token symbol
+ */
+function getTokenSymbol (token: string, eth: Eth, tokenAbi: AbiItem[] = ERC20Abi): Promise<SupportedTokens> {
+  if (token === ZERO_ADDRESS) {
+    return Promise.resolve('rbtc')
+  }
+
+  const contract = new eth.Contract(tokenAbi, token)
+  return contract.methods.symbol().call({ from: eth.accounts.create() })
```

Comment:
Ufff, so you are calling the ERC20 contract here to resolve for given token address its symbol here? I am not sure this is a good approach. How fast is this call? Also, how do you make sure that this is actually value defined in `SupportedTokens`? 

I would say first of all we want have defined list of tokens that we support (most probably already mentioned `config` should be good place to have it there) and since we gonna do that we can add their address as well? We already do this for every other interaction with blockchain, so I would say it is standard and it will saves us lot of this work, that I am not sure if actually bring many benefits.

Reply:
Hmm... The call is fast enough (it's static call), but I agree that tokens in the config would be better